gedcom/families.py

```diff
@@ -138,139 +125,149 @@ def validate(self):
                         "FAMILY",
                         key,
-                        family["id"],
+                        family.get_family_id(),
                         "NA",
-                        msg + wife["id"] + " " + wife["name"])
+                        msg + wife.get_person_id() + " " + wife.get_name())
 
     def _validate_death_of_parents_before_child_birth(self, family):
         """US09: validate death of parents before child birth
         """
         key = "US09"
         msg = "parent death before child birth for "
-        for child_id in family["children"]:
+        children = family.get_children()
+
+        for child_id in children:
             chil = self._people.individuals[child_id]
-            if family["husband_id"] is not None:
+            if family.get_husband_id() is not None:
                 # check the husband died after conception of child
-                husb = self._people.individuals[family["husband_id"]]
-                if husb["death_date"] is not None:
-                    hub9_date = husb["death_date"]
+                husb = self._people.individuals[family.get_husband_id()]
+                if husb.get_death_date() is not None:
+                    hub9_date = husb.get_death_date()
                     # Calculate 9 Months Back
                     for i in range(0, 9):
                         hub9_date = hub9_date.replace(day=1)
                         hub9_date = hub9_date - timedelta(days=1)
                         # Calculate Day
-                    if hub9_date.day > husb["death_date"].day:
+                    if hub9_date.day > husb.get_death_date().day:
                         hub9_date = hub9_date.replace(
-                            day=husb["death_date"].day)
-                    if hub9_date < chil["birth_date"]:
+                            day=husb.get_death_date().day)
+                    if hub9_date < chil.get_birth_date():
                         # error husband died at least 9 months before child birth
                         self._msgs.add_message(
                             "FAMILY",
                             key,
-                            family["id"],
+                            family.get_family_id(),
                             "NA",
-                            msg + husb["id"] + " " + husb["name"])
-            if family["wife_id"] is not None:
+                            msg + husb.get_person_id() + " " + husb.get_name())
+            if family.get_wife_id() is not None:
                 # check the wife died before the child birth
-                wife = self._people.individuals[family["wife_id"]]
-                if wife["death_date"] is not None and chil["birth_date"] > wife["death_date"]:
+                wife = self._people.individuals[family.get_wife_id()]
+                if wife.get_death_date() is not None and chil.get_birth_date() > wife.get_death_date():
                     # error wife died before divorce
```

Comment:
Can you fix this comment....should say
error wife died before child birth

Reply:
Fixed in latest version